community/rfcs/24-05-24-001-OPEA-Code-Structure.md

```diff
@@ -0,0 +1,66 @@
+**Author**
+
+[ftian1](https://github.com/ftian1), [lvliang-intel](https://github.com/lvliang-intel), [hshen14](https://github.com/hshen14)
+
+**Status**
+
+Under Review
+
+**Objective**
+
+Define a clear criteria and rule of adding new codes into OPEA projects.
+
+**Motivation**
+
+OPEA project consists of serveral repos, including GenAIExamples, GenAIInfra, GenAICompos, and so on. We need a clear definition on where the new code for a given feature should be put into for a consistent and well-orgnized code structure.
+
+
+**Design Proposal**
+
+The proposed code structure of GenAIInfra is:
+
+```
+GenAIInfra/
+├── kubernetes-addon/ # the folder implementing additional operational capabilities to Kubernetes applications
+├── microservices-connector/ # the folder containing the implementation of microservice connector on Kubernetes
+└── scripts/
+```
+
+The proposed code structure of GenAIExamples is:
+
+```
+GenAIExamples/
+└── ChatQnA/
+    ├── kubernetes/   # for deployment by kubernetes
+    │   ├── manifests/
+    │   ├── helm-charts/
+    │   └── microservices-connector/
+    ├── docker-composer/ # for deployment by docker-composer
```

Comment:
Let us rename this directory as "docker" versus docker-compose, a peer of the Kubernetes directory.